### ISSUE TITLE
multicore-sys-monitor@ccadeptic23 stops working after upgrade from mint 18.1 to 18.2

### ISSUE THREAD

**Issue by admzzz** [linuxmint/cinnamon-spices-applets#971]
applet could not be added from the panel's menu, the errors are:

error t=2017-07-03T01:55:34.800Z Applet is not defined
trace t=2017-07-03T01:55:34.817Z 
<----------------
@/home/admz/.local/share/cinnamon/applets/multicore-sys-monitor@ccadeptic23/applet.js:60:2
Extension.prototype._init@/usr/share/cinnamon/js/ui/extension.js:155:13
Extension@/usr/share/cinnamon/js/ui/extension.js:116:5
loadExtension@/usr/share/cinnamon/js/ui/extension.js:449:25
onEnabledAppletsChanged@/usr/share/cinnamon/js/ui/appletManager.js:236:13
---------------->
error t=2017-07-03T01:55:34.818Z [Applet "multicore-sys-monitor@ccadeptic23"]: Error importing applet.js from multicore-sys-monitor@ccadeptic23
error t=2017-07-03T01:55:34.818Z Could not load applet multicore-sys-monitor@ccadeptic23

**but after manual download and extraction to the ~/.local/share/cinnamon/applets it could be added to the panel and works ok.**, but it's menu ugly repeats Preferencies 9 times.

**Comment by jaszhix** [comment]
Since this applet has a legacy version you have to re-install version 1.6 after upgrading to Cinnamon 3.4.

**Comment by admzzz** [comment]
you erroneously closed this issue: re-installation from panel's menu returns an error (applet can not be added to the panel), only manual installation works.

**Comment by jaszhix** [comment]
Okay, give me your:
- Distribution
- Cinnamon version
- Applet version
- ~/.xsession-errors log when you try to add the applet

**Comment by admzzz** [comment]
$cat /etc/os-release
NAME="Linux Mint"
VERSION="18.2 (Sonya)"
ID=linuxmint
ID_LIKE=ubuntu
PRETTY_NAME="Linux Mint 18.2"
VERSION_ID="18.2"
HOME_URL="http://www.linuxmint.com/"
SUPPORT_URL="http://forums.linuxmint.com/"

$cinnamon --version
Cinnamon 3.4.3

Multi-Core System Monitor v 1.6.0 (from menu about, do not know how to retrieve it from command line)
BUG_REPORT_URL="http://bugs.launchpad.net/linuxmint/"
VERSION_CODENAME=sonya
UBUNTU_CODENAME=xenial

**Comment by jaszhix** [comment]
Try uninstalling it from the applet settings, delete ```~/.local/share/cinnamon/applets/multicore-sys-monitor@ccadeptic23``` and ```~/.cinnamon/configs/multicore-sys-monitor@ccadeptic23```. Then reinstall from Applet Settings, so it re-downloads it. It works for me.

**Comment by admzzz** [comment]
i think, that such clearance had to be done by the panel's (or applet's, dunno) scripts of uninstalling, not by the user. ;-)
have a good day.

**Comment by jaszhix** [comment]
I agree, but since the prefs.json file was moved to a location it doesn't get deleted on every upgrade (#854), this needs to be done one time, and then you shouldn't need to do it again.

**Comment by mikehaertl** [comment]
@jaszhix Thanks for your tip, this worked. But now there's still a red exclamation mark in the list of installed applets. When hovering it shows a warning, that this extension uses system calls that my slow down or freeze my system. You think, this can be fixed?

**Comment by admzzz** [comment]
jaszhix, mikehaertl,
had to add  (beside exclamation mark's problem) that if after using jaszhix's solution you switch to the tab All applets in the applets panel, you'll see that multicore-sys-monitor has to be upgraded, and after upgrade you'll be in the beginning situation: apllet does not work again. ;-)

**Comment by jaszhix** [comment]
@mikehaertl Code in the Cinnamon repository flags any applets using certain APIs through string matching. In the case of most applets, it isn't really accurate. Its a Cinnamon issue.

@admzzz Not following. The upgrade is bumpy going from Cinnamon 3.2 to 3.4. Once you're on 3.4, what you're describing is not the case.

**Comment by admzzz** [comment]
jaszhix,
ok let it describe that way (though it should be unknow to the upgrading end-user), can you find/make any workaround? it'll catch a lot of people during upgrade to mint 18.2 (btw, mint is the most used in the linux world distro).

**Comment by jaszhix** [comment]
I don't think deleting a couple folders for a compatibility breakage is a huge issue considering the app's handling of settings was a mess.  If I didn't patch it for 3.4 it would have stayed broken. I also have finite time and I cannot justify patching the legacy versions when the fix is that simple.

**Comment by sagarbehere** [comment]
I just upgraded Linux Mint from 18.1 to 18.2. Multi-core-system-monitor stopped working. Tried following @jaszhix recommendation of deleting the applet from the panel settings, removing the two folders, and re-installing from the applet settings. After deleting the applet from the panel settings, I could no longer see the two directories mentioned. It seems like they were removed when the applet was deleted from the panel settings. I then went ahead and installed the applet again. It downloaded the applet anew and installed it. All of that did not work, I continue to get the same error message. So there is at least one case (mine) where the instructions are not working to fix the issue. How to proceed?

**Comment by jaszhix** [comment]
Make sure you are refreshing the applet list in xlet settings after removing the applet.

Is it creating ```~/.cinnamon/configs/multicore-sys-monitor@ccadeptic23```, and if so is prefs.json located there?

Do you see any errors in ```~/.cinnamon/glass.log``` or ```~/.xsession-errors```?

**Comment by sagarbehere** [comment]
Ahh.. refreshing the applet list prior to reinstalling it did the trick. Thank you. The matter is now resolved.

For reference, the full steps when upgrading from Linux Mint 18.1 were

1. Go to applet settings and remove/delete the multi-core-system-monitor applet
2. Verify that the two directories mentioned above are already removed when the applet is deleted. This was the case for me, but I guess if they are still around, they can be deleted manually
3. Go the the applet list and refresh
4. Reinstall.

Thank you very much :)

**Comment by geocohn** [comment]
On my 2 Ubuntu 16.04 systems at home I followed @jaszhix's advice to delete the directories and reinstall, and it worked great. However on my similar system at work, it failed no matter what I tried, saying it couldn't find applet.js. I finally figured out my stock 16.04 system had a downrev version of Cinnamon. I upgraded by adding the [unofficial PPA](https://launchpad.net/~embrosyn/+archive/ubuntu/cinnamon) and that did the trick:
```
sudo add-apt-repository ppa:embrosyn/cinnamon
sudo apt update
sudo apt full-upgrade
```
